### DISCUSSION TITLE
Add bookmarks

### DISCUSSION THREAD

**Discussion by venteto** [bluesky-social/atproto#2000] [Bluesky Client App]
This seems like it would be almost as straigthforward as copying and pasting the code that implements Likes and then renaming the client menu entry?

**Comment by RobotoSkunk** [comment]
I'll bump this because it might be useful

**Comment by qazmlp** [comment]
Bookmarks are probably expected to be private. (Likes are public, just not surfaced.)
That means the implementation would have to be pretty different, conceptually more similar to that of mutes I think.

There's a feature request in the social-app repo here: https://github.com/bluesky-social/social-app/issues/1091

**Comment by davidcelis** [comment]
I posted a similar message in the linked issue (bluesky-social/social-app#1091) but I'll share it here as well, because I think bookmarks should be implemented in the protocol rather than the social app.

I agree that Bluesky needs native support for bookmarks; this is a feature that people have come to expect from social networks like this. The 📌 feed is an interesting workaround, but [this post](https://bsky.app/profile/pfrazee.com/post/3lb4bx7iguk2b) from @pfrazee makes me worry about support for _workarounds_ entering the codebase for the social app rather than doing the work on protocol-level support for bookmarks. I also think it has to be part of the protocol itself instead of just the social app because people will expect their bookmarks to be viewable from any device or from the web, meaning the app shouldn't be storing them as separate state.

Looking through the lexicons, I'm not sure it's as simple as duplicating Likes except making it a method on `.repo`; as far as I can tell, most `com.atproto.repo` procedures do not require auth, including `com.atproto.repo.getRecord`. I may be misunderstanding whether or not repo records can truly be public or private, but it seems that all records are meant to be public.

I also agree that duplicating _mutes_ makes more sense than duplicating likes. As far as I can tell, most `com.atproto.repo` procedures do not require auth, including `com.atproto.repo.getRecord`. I may be misunderstanding whether or not repo records can truly be public or private, but it seems like all records are meant to be public? 

While I was looking through the existing lexicons, I wasn't sure that there was any existing package within the `app.bsky` namespace that makes sense for bookmarks. Mutes are in `app.bsky.graph`, but bookmarks aren't really a social graph concern. I thought `app.bsky.actor` or `com.atproto.server` _might_ make sense, but the former seems very focused on onboarding and user settings (and people would probably argue, correctly, that bookmarks aren't settings), and the latter seems very focused on credentials.

Maybe we need a new small `app.bsky.bookmark` package for this that follows the similar "require auth, only create and enumerate records for the actor" pattern present for mutes?

**Comment by ebwinters** [comment]
The initial work at the lexicon and now client level is being done by myself and members of the lexicon-community org.
[Here](https://github.com/lexicon-community/lexicon/pull/12) is the recently merged lexicon
[Here](https://github.com/lexicon-community/js-lexicon-community-types/pull/2) is a draft of a client package for consumption in Bluesky or other app views

I could use some help doing the UI implementation in Bluesky, but the POC is done 
![image](https://github.com/user-attachments/assets/db0cfd14-d1ae-4788-8807-33babe338e63)

**Reply by danielbayley** [reply]
> I could use some help doing the UI implementation in Bluesky, but the POC is done

@ebwinters Since a new bookmark icon will be needed, to match [these](https://github.com/bluesky-social/social-app/tree/main/assets/icons) as part of this, I threw up a quick PR. See https://github.com/bluesky-social/social-app/pull/7231. (Full disclosure, I'm a core contributor to [_Lucide_ icons](https://lucide.dev)…)

**Comment by ebwinters** [comment]
PR is out: https://github.com/bluesky-social/social-app/pull/7383

**Comment by mozzius** [comment]
Done!
